fish-rust/build.rs

```diff
@@ -131,3 +145,59 @@ fn detect_bsd() -> miette::Result<bool> {
     assert!(result, "Target incorrectly detected as not BSD!");
     Ok(result)
 }
+
+/// Detect libintl/gettext and its needed symbols to enable internationalization/localization
+/// support.
+fn have_gettext(detector: &Detector) -> miette::Result<bool> {
+    // The following script correctly detects and links against gettext, but so long as we are using
+    // C++ and generate a static library linked into the C++ binary via CMake, we need to account
+    // for the CMake option WITH_GETTEXT being explicitly disabled.
+    println!("cargo:rerun-if-env-changed=CMAKE_WITH_GETTEXT");
+    if let Some(with_gettext) = std::env::var_os("CMAKE_WITH_GETTEXT") {
+        println!(
+            "cargo:warning=CMAKE_WITH_GETTEXT={}",
+            with_gettext.to_string_lossy()
+        );
+        if with_gettext.eq_ignore_ascii_case("0") {
+            println!("cargo:warning=gettext disabled in CMake");
+            return Ok(false);
+        }
+    }
+
+    // In order for fish to correctly operate, we need some way of notifying libintl to invalidate
+    // its localizations when the locale environment variables are modified. Without the libintl
+    // symbol _nl_msg_cat_cntr, we cannot use gettext even if we find it.
+    let mut libraries = Vec::new();
+    let mut found = 0;
+    let symbols = ["gettext", "_nl_msg_cat_cntr"];
+    for symbol in &symbols {
+        // Historically, libintl was required in order to use gettext() and co, but that
+        // functionality was subsumed by some versions of libc.
+        if detector.has_symbol_in::<&str>(symbol, &[]) {
+            // No need to link anything special for this symbol
+            found += 1;
+            println!("cargo:warning=Found symbol {symbol} in standard library");
+            continue;
+        }
+        for library in ["intl", "gettextlib"] {
+            if detector.has_symbol(symbol, library) {
+                libraries.push(library);
+                println!("cargo:warning=Found symbol {symbol} in lib{library}");
+                found += 1;
+                continue;
+            }
+        }
+    }
+    match found {
+        0 => {
+            println!("cargo:warning=gettext not found");
```

Comment:
Please do not introduce this warning by default. I don't need to see this every time I build. We should only emit this warning if the user explicitly requests gettext and it's not found.

Reply:
This was for debugging for myself as I tried to figure this out.